src/main/java/bio/terra/cli/service/utils/HttpUtils.java

```diff
@@ -271,23 +271,28 @@ public static HttpResponse sendHttpRequest(
    * <p>- Retry the request to add a user to a workspace.
    *
    * @param makeRequest function to perform the request
+   * @param makeRequestIsRetryable function to test whether an exception thrown by makeRequest is
+   *     retryable or not
    * @param isOneTimeError function to test whether the exception is the expected one-time error
    * @param handleOneTimeError function to handle the one-time error before retrying the request
-   * @param isRetryable function to test whether the exception is retryable or not
+   * @param handleOneTimeErrorIsRetryable function to test whether an exception thrown by
+   *     handleOneTimeError is retryable or not
    * @param <T> type of the response object (i.e. return type of the makeRequest function)
    * @return the response object
    * @throws E1 if makeRequest throws an exception that is not the expected one-time error
    * @throws E2 if handleOneTimeError throws an exception
    */
-  public static <T, E1 extends Exception, E2 extends Exception> T callAndHandleOneTimeError(
-      SupplierWithCheckedException<T, E1> makeRequest,
-      Predicate<Exception> isOneTimeError,
-      SupplierWithCheckedException<Void, E2> handleOneTimeError,
-      Predicate<Exception> isRetryable)
-      throws E1, E2, InterruptedException {
+  public static <T, E1 extends Exception, E2 extends Exception>
+      T callAndHandleOneTimeErrorWithRetries(
```

Comment:
Since you frequently call this with the same predicates like `samService::isRetryable`, you could make shorter overloads and use them when the defaults apply.

Reply:
Good idea. Added an overload just for `SamService` since all 3 occurrences there are using the same `isRetryable` method.